### DISCUSSION TITLE
"Days to expire" with minus when set expiration day to object in bucket.

### DISCUSSION THREAD

**Discussion by atominkiss** [minio/minio#16417] [Q&A]
Good day.
I'm using:
mc version RELEASE.2023-01-11T03-14-16Z (commit-id=14c2e506fa78b53fb6db88bcf87d8f6d3fb6989e)
Runtime: go1.19.4 linux/amd64
Copyright (c) 2015-2023 MinIO, Inc.
License GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>

when i set expiration to bucket using mc and this command:

**mc ilm import minio/102938-4-300 <<EOF
{
    "Rules": [
        {
            "Expiration": {
                "Date": "2023-02-16T00:00:00Z"
            },
            "ID": "expire-bucket",
            "Status": "Enabled"
        }
    ]
}
EOF**
i see that expiration day with minus and i don't understand why...

**mc ilm rule ls minio/102938-4-300**
┌─────────────────────────────────────────────────────────┐
│ Expiration for latest version (Expiration)                                     │
├───────────────   ┬─────────    ┬───────-------─┬──────----------------------┤
│ ID                     │ STATUS  │ PREFIX │ TAGS │ DAYS TO EXPIRE │ EXPIRE DELETEMARKER │
├──────────────   ─┼─────────   ┼───────         ─┼──────----------------------┤
│ expire-bucket │ Enabled│ -            │ -         │       **-30**          │-     false                             │
└───────────────   ┴─────────   ┴────────         ┴──────----------------------┘

**Comment by vadmeste** [comment]
@atominkiss, this looks like an issue in mc rendering the table. I was able to reproduce. I will send a fix shortly.

**Reply by vadmeste** [reply]
@atominkiss in fact, it turned out this is expected. This means there is still at least 30 more days until the ILM be executed

**Reply by atominkiss** [reply]
@vadmeste when I change the date to:
**mc ilm import minio/102938-4-300 <<EOF
{
    "Rules": [
        {
            "Expiration": {
                "Date": "2023-02-01T00:00:00Z"
            },
            "ID": "expire-bucket",
            "Status": "Enabled"
        }
    ]
}
EOF**
i catch this:
** mc ilm rule ls minio/102938-4-300**
┌────────────────────────────────────────────────────────────────────────────────┐
│ Expiration for latest version (Expiration)                                     │
├───────────────┬─────────┬────────┬──────┬────────────────┬─────────────────────┤
│ ID            │ STATUS  │ PREFIX │ TAGS │ DAYS TO EXPIRE │ EXPIRE DELETEMARKER │
├───────────────┼─────────┼────────┼──────┼────────────────┼─────────────────────┤
│ expire-bucket │ Enabled │ -      │ -    │            -15 │ false               │
└───────────────┴─────────┴────────┴──────┴────────────────┴─────────────────────┘
Still see minus, but expires less than 30 days.

**Reply by vadmeste** [reply]
@atominkiss yes, this is a little bit confusing, just a user experience thing, but it will work as expected.

/cc @krisis, maybe this should show the actual expiry date, it looks confusing right now

**Reply by atominkiss** [reply]
@vadmeste 
there is another one thing when i set lifecycle thrue mc. In web form i don't see correct data.  Look at screenshot.
![image](https://user-images.githubusercontent.com/27758671/212656040-e7f75488-3253-43fb-aa6d-da8e07d365fc.png)

**Reply by harshavardhana** [reply]
@atominkiss please open relevant issues on each repo with a reproducer - thank you.
